src/charm.py

```diff
@@ -101,34 +103,13 @@ def mongodb_config(self) -> MongoDBConfiguration:
         hosts = [self.get_hostname_by_unit(self.unit.name)] + [
             self.get_hostname_by_unit(unit.name) for unit in peers.units
         ]
-        external_ca, _ = self.tls.get_tls_files("unit")
-        internal_ca, _ = self.tls.get_tls_files("app")
-
-        return MongoDBConfiguration(
-            replset=self.app.name,
-            database="admin",
-            username="operator",
-            password=self.get_secret("app", "operator_password"),
-            hosts=set(hosts),
-            roles={"default"},
-            tls_external=external_ca is not None,
-            tls_internal=internal_ca is not None,
-        )
+        return self._get_mongodb_config_for_user(OperatorUser, set(hosts))
 
     @property
     def monitor_config(self) -> MongoDBConfiguration:
         """Generates a MongoDBConfiguration object for this deployment of MongoDB."""
-        return MongoDBConfiguration(
-            replset=self.app.name,
-            database="",
-            username="monitor",
-            # MongoDB Exporter can only connect to one replica - not the entire set.
-            password=self.get_secret("app", "monitor_password"),
-            hosts={self.get_hostname_by_unit(self.unit.name)},
-            roles={"monitor"},
-            tls_external=self.tls.get_tls_files("unit") is not None,
-            tls_internal=self.tls.get_tls_files("app") is not None,
-        )
+        hosts = set([self.get_hostname_by_unit(self.unit.name)])
```

Comment:
Same comment about casting to `Set`, considering changing the method to accept a list.

Reply:
make sense, updated